### PR TITLE
Integration

### PR DESCRIPTION
**Pull request by alexmedinasf**
For this milestone of the project, as the driver, and Dani as the navigator we:

Solve the N+1 problem when fetching all posts and their comments for a user

Used Capybara to write integration tests for each view present in the project.

[Link to the screenshots](https://imgur.com/a/agYBH88)

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -7,6 +7,7 @@ def index
   def show
     @post = Post.find(params[:id])
     @user = User.find(@post.author_id)
+    @comments = @post.comments.includes(:author)
```
Comment (ShadyShawkat) [inline app/controllers/posts_controller.rb]: - [ ] As per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=fetching%20all%20posts%20and%20their%20comments%20for%20a%20user) requirement, please make sure to fix the N+1 problem when fetching all posts and their comments for a user which can be done using the following:

```ruby
@posts = @user.posts.includes(:comments)
```

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @alexmedinasf, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (alexmedinasf) [reply app/controllers/posts_controller.rb]: Thank you for the tip, @ShadyShawkat ! I'll do the change right now!

**Review by cynthiainga — APPROVED** [review]
Hi @alexmedinasf, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] Requested change is made ✔️ 
- [ ] No linters errors ✔️ 
- [ ] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**